### PR TITLE
MAINT: Remove unused arguments to LogisticRegression SPMD

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR removes unused arguments to the SPMD version of logistic regression, as it ignores some of them.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Failures do not seem to be blamable on this PR.

**Comment by avolkov-intel** [comment]
I think we can hard-code method "dense_batch" in this line here https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/linear_model/logistic_regression.py#L66 and avoid having this argument at all

**Comment by david-cortes-intel** [comment]
> I think we can hard-code method "dense_batch" in this line here https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/linear_model/logistic_regression.py#L66 and avoid having this argument at all

Would it require removing the argument from some other function signature?

**Comment by avolkov-intel** [comment]
> > I think we can hard-code method "dense_batch" in this line here https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/linear_model/logistic_regression.py#L66 and avoid having this argument at all
> 
> Would it require removing the argument from some other function signature?

I'm pretty sure it only should be removed from class declaration in this file, it shouldn't be used in any other places

https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/linear_model/logistic_regression.py

**Comment by david-cortes-intel** [comment]
> This is impacting beyond spmd, it is impacting all onedal GPU use of logisticregression please reflect that in the title and description.

I don't follow. The non-SPMD GPU estimator from sklearnex would already check for conditions that wouldn't result in passing what's removed here. Only the SPMD estimator would allow calling it with all the arguments it offers that it then ignores.

> As per the evolving design documents, these would be sklearn conformance additions, and are in the wrong module. Thanks for removing them.

Which design documents do you mean?

**Comment by david-cortes-intel** [comment]
Looks like these arguments were actually needed for some reason for the non-SPMD GPU version to work, so not sure what would be the best way to remove them from SPMD instead of accepting but ignoring.
